PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/runner/deploy/SpringCloudDeploymentState.java

```diff
@@ -175,142 +157,40 @@ protected String getDeployTarget() {
 
     @Override
     protected void updateTelemetryMap(@NotNull Map<String, String> telemetryMap) {
-        telemetryMap.putAll(springCloudDeployConfiguration.getModel().getTelemetryProperties());
+        telemetryMap.putAll(config.getModel().getTelemetryProperties());
     }
 
-    private void validateSpringCloudAppArtifact(String finalJar) throws AzureExecutionException, IOException {
-        final JarFile jarFile = new JarFile(finalJar);
-        final Attributes manifestAttributes = jarFile.getManifest().getMainAttributes();
-        final String mainClass = manifestAttributes.getValue(MAIN_CLASS);
-        if (StringUtils.isEmpty(mainClass)) {
-            throw new SpringCloudValidationException(String.format(MAIN_CLASS_NOT_FOUND, finalJar));
-        }
-        final String library = manifestAttributes.getValue(SPRING_BOOT_LIB);
-        if (StringUtils.isEmpty(library)) {
+    private void printPublicUrl(final SpringCloudApp app, @NotNull RunProcessHandler processHandler) {
```

Comment:
RunProcessHandler  should be replaced with more generate interface like IProgressReport

Reply:
https://dev.azure.com/mseng/VSJava/_workitems/edit/1812811